transportation_protocols/tcp.py

```diff
@@ -0,0 +1,35 @@
+import socket
+
+
+class TCP_Connection:
+
+    local_ip = "127.0.0.1"
+    
+    def __init__(self, destination_port, is_client, destination_ip=local_ip):
+        self._destination_ip = destination_ip
+        self._destination_port = destination_port
+        if is_client:
+            self._tcp_connection = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+            self._tcp_connection.connect((self._destination_ip, self._destination_port))
+        else:
+            connection_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+            connection_socket.bind((self._destination_ip, self._destination_port))
```

Comment:
The names destination ip / destination port are a bit problematic.
for clients it's fine, but for servers it is being used here as a **local** ip + port.

Reply:
understood, changed it according to the connection created in the connection factory